platform_channels/lib/src/platform_image_demo.dart

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:platform_channels/src/image_basic_message_channel.dart';
+
+/// Demonstrates how to use [BasicMessageChannel] to get an image from platform.
+///
+/// The widget uses [Image.memory] to display the image obtained from the
+/// platform.
+class PlatformImageDemo extends StatefulWidget {
+  @override
+  _PlatformImageDemoState createState() => _PlatformImageDemoState();
+}
+
+class _PlatformImageDemoState extends State<PlatformImageDemo> {
+  Future<Uint8List> imageData;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Platform Image Demo'),
+      ),
+      body: Builder(
```

Comment:
nit: Now that you're not using a snackbar, you don't need this builder any more.